tests/foreman/cli/test_repository.py

```diff
@@ -983,6 +989,166 @@ def test_positive_upload_content(self):
             result[0]['message'],
         )
 
+    @skip_if_bug_open('bugzilla', 1410916)
+    @run_only_on('sat')
+    @tier2
+    def test_negative_restricted_user_cv_add_repository(self):
+        """Attempt to add a product repository to content view with a
+        restricted user, using product name not visible to restricted user.
+
+        @id: 65792ae0-c5be-4a6c-9062-27dc03b83e10
+
+        @BZ: 1410916
+
+        @Steps:
+            1. Setup a restricted user with the permissions that restrict
+               to view the products with names like Test_* or "rhel7*"
+            2. Create a content view
+            3. Create a product with name that should not be visible to the
+               user and add a repository to it
+            4. Create a content view
+
+        @Assert:
+            1. The admin user can view the product repository
+            2. The restricted user cannot view the product repository
+            3. The restricted user cannot add the product repository to a
+               content view
+            4. After the attempt of adding the product repository to content,
+               assert that the restricted user still cannot view the product
+               repository.
+
+        @CaseLevel: Integration
+        """
+        required_permissions = {
+            'Katello::Product': (
+                [
+                    'view_products',
+                    'create_products',
+                    'edit_products',
+                    'destroy_products',
+                    'sync_products',
+                    'export_products'
+                ],
+                'name ~ "Test_*" || name ~ "rhel7*"'
```

Comment:
How about defining local variable called `pattern` with random value instead of "Test"? Such hardcoded values can become a headache if we decide to drop extra org creation at some point (as extra orgs are impacting badly on performance).